lesson_64/HW_L64/server.js

```diff
@@ -33,6 +43,95 @@ const getOrderByUserId = (userId) => {
   return orders.find((order) => order.userId === userId);
 };
 
+const readProductsStore = async (filename) => {
+  try {
+    const data = await readFile(`${filename}`, { encoding: "utf8" });
+    return JSON.parse(data);
+  } catch (err) {
+    throw new ErrorReadWriteFile(err);
+  }
+};
+
+const writeProductsStore = async (filename, data) => {
+  try {
+    await writeFile(`${filename}`, JSON.stringify(data), { encoding: "utf8", flag: "w" });
+  } catch (err) {
+    throw new ErrorReadWriteFile(err);
+  }
+};
+
+const getCustomProductById = (productId, productsList) => {
+  return productsList.find((product) => product.id === productId);
+};
+
+const ensureDirectoryExists = (directory) => {
+  if (!existsSync(directory)) {
+    mkdirSync(directory);
+  }
+};
+
+const ensureFileExists = (filename) => {
+  if (!existsSync(filename)) {
+    throw new ErrorObjectNotFound('File not found');
+  }
+};
+
+const handleFileUpload = async (req, res, next, uploadParams) => {
+  const {
+    productId,
+    filename,
+    previewFilename=null,
+    filePath,
+    previewFilePath=null,
+    fileType
+  } = uploadParams;
+
+  eventEmitter.emit('fileUploadStart', {productId, filename});
+  const writeableStream = createWriteStream(filePath, { encoding: "binary", flags: "w" });
+  req.pipe(writeableStream)
+    .on('finish', async () => {
+      try {
+        const customProductsList = await readProductsStore(productsStore);
+        const foundProduct = getCustomProductById(`${productId}`, customProductsList);
```

Comment:
I would suggest to convert the first parameter to string inside the `getCustomProductById` function because with the current approach, you have to do it every time you call a function.